packages/cw1/src/query.rs

```diff
@@ -21,5 +21,5 @@ where
 
 #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
 pub struct CanExecuteResponse {
-    pub can_execute: bool,
+    pub can_execute: Vec<bool>,
```

Comment:
Do we really want this return type? Woudn't it be sufficient and simpler if we return the && combined response? What do you think, @ethanfrey?

Reply:
The return result `[true, true, false]` does not have precise information about which message has crashed the tx. Instead of `Vec<bool>` I would recommend returning a `StdResult<ContractErr>`

Reply:
> The return result [true, true, false] does not have precise information about which message

Oh, it does. The order of the results is the same as the order of submitted messages.

> simply returning a StdResult<ContractErr>

Such an error is raised when trying to execute the messages. For this query both _ok_ and _not ok_ should not be error cases.

Reply:
> Do we really want this return type?

It's useful for testing, as you can see below. Besides that, I'm OK with both options... I'm not sure how useful this really can be for writing contracts / clients.

Reply:
Another (related) thing, is that that struct can be moved to the cw1 package (it's currently being defined in each contract). If you want I can do that here or in another PR

Reply:
Which structure? `CanExecuteResponse` is in `packages/cw1/src/query.rs`

Reply:
You're right... when making the changes I was left under the impression `query.rs` was defined in both cw1 contracts.